python/ray/data/dataset.py

```diff
@@ -646,7 +646,7 @@ def process_next_batch(batch: DataBatch) -> Iterator[Block]:
 
         plan = self._plan.with_stage(
             OneToOneStage(
-                "map_batches",
+                getattr(fn, "__name__", type(fn)),
```

Comment:
In the case where we fall back to `type(fn)` (I assume for example, in the case of a lambda function), would including `"map_batches"` in the output name be helpful to users?

Reply:
So the one time I can see where this comes up is not when you pass in a lambda function (`x = lambda y: y; x.__name__` returns `<lambda>`) but rather when you pass in an instance of a class with a `__call__` method defined. In this case `type(fn)` returns the class as a string.

@bveeramani @clarkzinzow @amogkam What do you prefer in this case? Should I use something like `f'map_batches: {type(fn)}'` as a fallback?